applications/PfemFluidDynamicsApplication/python_scripts/remesh_fluid_domains_process.py

```diff
@@ -164,21 +166,24 @@ def ExecuteInitializeSolutionStep(self):
         currentTime=self.main_model_part.ProcessInfo[KratosMultiphysics.TIME]
         currentStep=self.main_model_part.ProcessInfo[KratosMultiphysics.STEP]                
 
-        if(currentStep > 1):
+        if currentStep >= 2 and self.fileTotalVolume is None:
+            self.fileTotalVolume = open("totalVolumeBeforeMeshing.txt",'w')
+
+        if(currentStep > 1 and self.fileTotalVolume is not None):
             for domain in self.meshing_domains:
                 if(domain.Active()):
                     domain.ComputeAverageMeshParameters()  
```

Comment:
Be aware that this function is now not called unless the output file has been initialized. I don't know if this is a problem for you, but the behavior has changed in relation to the original version.